plugin/Record/RecordManagerDuplicateListViewForm.php

```diff
@@ -0,0 +1,192 @@
+					continue;
+				}
+				$newListView[$key] = $value;
+			}
+			$newListView = array_merge($newNameData, $newListView);
+
+			$this->_pWPDB->insert(
+				$tableListViews,
+				$newListView
+			);
+			$duplicateListViewId = $this->_pWPDB->insert_id;
+
+			if ($duplicateListViewId !== 0) {
+				//duplicate data related oo_plugin_fieldconfig table
+				$tableFieldConfig = $prefix . self::TABLENAME_FIELDCONFIG_FORMS;
+				foreach ($listViewRoot['fields'] as $field) {
+					$selectFieldConfigByIdAndFieldName = "SELECT * FROM {$this->_pWPDB->_escape($tableFieldConfig)} WHERE form_id='{$this->_pWPDB->_escape($id)}' AND fieldname ='{$this->_pWPDB->_escape($field)}'";
+					$fieldConfigRows = $this->_pWPDB->get_results($selectFieldConfigByIdAndFieldName);
+					if (!empty($fieldConfigRows) && (count($fieldConfigRows) !== 0)) {
+						foreach ($fieldConfigRows as $fieldConfigRow) {
+							$newFieldConfigRow = [];
+							$newFieldConfigRow['form_id'] = $duplicateListViewId;
+							$newFieldConfigRow['order'] = $fieldConfigRow->order;
+							$newFieldConfigRow['fieldname'] = $field;
+							$newFieldConfigRow['fieldlabel'] = $fieldConfigRow->fieldlabel;
+							$newFieldConfigRow['module'] = $fieldConfigRow->module;
+							$newFieldConfigRow['individual_fieldname'] = $fieldConfigRow->individual_fieldname;
+							$newFieldConfigRow['required'] = $fieldConfigRow->required;
+							$newFieldConfigRow['availableOptions'] = $fieldConfigRow->availableOptions;
+							$this->_pWPDB->insert($tableFieldConfig, $newFieldConfigRow);
+						}
+					}
+				}
+
+				//duplicate data related oo_plugin_fieldconfig_form_defaults table
+				$formDefaultsFieldConfigTable = $prefix . self::TABLENAME_FIELDCONFIG_FORM_DEFAULTS;
+				$formDefaultsFieldConfigValueTable = $prefix . self::TABLENAME_FIELDCONFIG_FORM_DEFAULTS_VALUES;
+				$selectFormDefaultsById = "SELECT * FROM {$this->_pWPDB->_escape($formDefaultsFieldConfigTable)} WHERE form_id='{$this->_pWPDB->_escape($id)}'";
+				$formDefaultsRows = $this->_pWPDB->get_results($selectFormDefaultsById);
+				if (!empty($formDefaultsRows) && (count($formDefaultsRows) !== 0)) {
+					foreach ($formDefaultsRows as $formDefaultsRow) {
+						$newFormDefaultsRow = [];
+						$newFormDefaultsRow['form_id'] = $duplicateListViewId;
+						$newFormDefaultsRow['fieldname'] = $formDefaultsRow->fieldname;
+						$this->_pWPDB->insert($formDefaultsFieldConfigTable, $newFormDefaultsRow);
+
+						$newDefaultsId = $this->_pWPDB->insert_id;
+						$selectFormDefaultValueById = "SELECT * FROM {$this->_pWPDB->_escape($formDefaultsFieldConfigValueTable)} WHERE defaults_id ='{$this->_pWPDB->_escape($formDefaultsRow->defaults_id)}'";
+						$formDefaultsValueRows = $this->_pWPDB->get_results($selectFormDefaultValueById);
+						if (!empty($formDefaultsValueRows) && (count($formDefaultsValueRows) !== 0)) {
+							foreach ($formDefaultsValueRows as $formDefaultsValueRow) {
+								$newFormDefaultsValueRow = [];
```

Comment:
The depth of indent screams for refactoring into smaller functions. Especially because copying rows is the same concept every time.